scripts/scons_plugin/config.py

```diff
@@ -457,7 +457,19 @@ def AddPkgConfigDependency(context, package, flags):
         context.Result('pkg-config not available')
         return False
 
-    cmd = '%s %s --silence-errors %s' % (quote(pkg_config), package, flags)
+
+    cmd = ''
+    if prefix:
+        pkg_config_path = os.path.join(prefix, 'lib', 'pkgconfig')
+        global_pkg_config_path = env.get('PKG_CONFIG_PATH', None)
+
+        # set user-defined $(prefix)/lib/pkgconfig path for pkg-config if needed (lowest
+        # priority); otherwise PKG_CONFIG_PATH is propagated implicitly
+        if os.path.isdir(pkg_config_path):
+            if global_pkg_config_path:
+                pkg_config_path = global_pkg_config_path + ':' + pkg_config_path
+            cmd = 'env PKG_CONFIG_PATH={} '.format(quote(pkg_config_path))
+    cmd += '{} {} --silence-errors {}'.format(quote(pkg_config), package, flags)
```

Comment:
I just realized that we had a bug (before this PR): we did not propagate PKG_CONFIG_PATH to env.

If user set PKG_CONFIG_PATH through env var, it's not needed, but if user did it via scons arguments, like this:

```
scons PKG_CONFIG_PATH=/path
```

then we wont pass it to pkg-config, but should.

Could you rework this block to also pass PKG_CONFIG_PATH to cmd when `prefix` is not set, but `global_pkg_config_path` is set?

Reply:
done